cheddar/cheddar-server/src/main/java/com/clicktravel/cheddar/server/http/filter/aws/AWSXrayRequestFilter.java

```diff
@@ -45,46 +52,55 @@
+                final String segmentName = applicationConfiguration.name();
+                final String amazonTraceId = requestContext.getHeaderString("X-Amzn-Trace-Id");
+
+                if (shouldSampleSegment(segmentName, requestContext)) {
+                    Segment segment;
+                    if (amazonTraceId != null) {
+                        final TraceID awsXrayTraceId = TraceID.fromString(amazonTraceId);
+                        segment = AWSXRay.beginSegment(segmentName, awsXrayTraceId, null);
+                    } else {
+                        segment = AWSXRay.beginSegment(segmentName);
+                    }
+                    addRequestParamsToSegment(segment, requestContext);
+                    AWSXraySegmentContextHolder.set(new DefaultAWSXraySegmentContext(segment));
+                }
+            } catch (final Exception e) {
+                logger.debug("Failed to begin XRay segment due to exception " + e.getMessage());
             }
-            final Segment segment = AWSXRay.beginSegment(segmentName);
-            AWSXraySegmentContextHolder.set(new DefaultAWSXraySegmentContext(segment));
-        } catch (final Exception e) {
-            logger.debug("Failed to begin XRay segment due to exception " + e.getMessage());
         }
     }
 
-    /**
-     * In order to better group the XRay segments we would like name them after the request's path without its template
-     * params substituted for the real values and no braces e.g GET wire/id/queue.
-     *
-     * @param requestMethod the string verb used for the request e.g. POST
-     * @param requestUriInfo for the current request
-     * @return A String that represents the path with template params.
-     */
-    private String getPathWithTemplateParamsNotValues(final String requestMethod,
-            final ExtendedUriInfo requestUriInfo) {
-        final List<UriTemplate> matchedTemplates = requestUriInfo.getMatchedTemplates();
-        final StringBuilder builder = new StringBuilder();
-
-        builder.append(requestMethod);
-        builder.append(" ");
-
-        for (int i = matchedTemplates.size() - 1; i >= 0; i--) {
-            builder.append(matchedTemplates.get(i).getTemplate().replaceAll("\\{", "").replaceAll("\\}", ""));
-        }
+    private boolean shouldSampleSegment(final String segmentName, final ContainerRequestContext requestContext) {
+        final SamplingRequest samplingRequest = new SamplingRequest(segmentName,
+                requestContext.getUriInfo().getBaseUri().getHost(), requestContext.getUriInfo().getPath(),
+                requestContext.getMethod(), requestContext.getHeaderString("Origin"));
+        final SamplingStrategy chosenSamplingStrategy = AWSXRay.getGlobalRecorder().getSamplingStrategy();
+        // As we have not changed the sampling strategy so XRay should call out to AWS console for settings
+        final SamplingResponse sample = chosenSamplingStrategy.shouldTrace(samplingRequest);
```

Comment:
does this actually aws each time?

Reply:
nvm i read the linked doc, look good

Reply:
Not that I can see, it seems that there is some polling strategy thing in place 

``` 
if (!isStarted) {
       startPoller();
}
  SamplingResponse sampleResponse;
  .......
    for (CentralizedRule rule : manifest.getRules().values()) {
            boolean applicable = rule.match(samplingRequest);
  ......
 CentralizedRule dRule = manifest.getDefaultRule();
 if (dRule != null) {
            logger.debug("Applicable default rule: " + dRule.getName());
            return dRule.sample(Instant.now());
   }
```

Seems to drop through to that return locally, but from what I can tell it matches that of the engineering env rules....although it is a little hard to tell 😬 I'm hoping it'll be clearer in UAT.

Reply:
So I think it goes a fetches the rules into memory and then calls out every so often and updates them.

Reply:
yeah its not clear that doc sort half suggests that the daemon should do the strategy work rather than the app choosing to sample or not, but lets go with this

Reply:
I actually only added it because if you use the AWSXRayServletFilter has code similar (more advanced) to this so I thought as nothing has been simple with this so far, explicitly executing against the sample strategy seems logical and hopefully it should be using the same as the daemon so it _shouldn't_ make a different if it is 🤞